syntax = "proto3";
package protos;



// 角色快照数据
message role_snapshot_t
{
    int64 role_id = 1;      // 角色id
    string nickname  = 2;   // 昵称
}

// 玩家游戏角色数据
message role_game_data
{
	int64 role_id = 1;
	string nickname = 2;
	int32 zone_id = 3;
}


// E_LOGIN_MSG_C2S_REGISTER_ACCOUNT                    = 1,  // 注册账号
message req_E_LOGIN_MSG_C2S_REGISTER_ACCOUNT
{
    string account_name = 1;    // 账号名字
    string pswd = 2;            // 密码
}

message rep_E_LOGIN_MSG_C2S_REGISTER_ACCOUNT
{
    int32 isok = 1;
    string account_name = 2;    // 账号名字
    string pswd = 3;            // 密码
    int64 account_id = 4;       // 账号唯一标识id
}

// E_LOGIN_MSG_C2S_LOGIN                               = 2,  // c->s登录
message req_E_LOGIN_MSG_C2S_LOGIN
{
    string account_name = 1;    // 账号名字
    string pswd = 2;            // 密码
    int64 account_id = 3;       // 账号唯一id
    int32 zone_id = 4;          // 区号id
}

message rep_E_LOGIN_MSG_C2S_LOGIN
{
    int32 isok = 1;
    repeated role_snapshot_t role_list = 2; // 角色快照列表
}

// E_LOGIN_MSG_GG2ROOT_LOGIN                           = 3,  // gg->root请求登录
message req_E_LOGIN_MSG_GG2ROOT_LOGIN
{
    string account_name = 1;    // 账号名字
    string pswd = 2;            // 密码
    int64 account_id = 3;       // 账号唯一id
    int32 zone_id = 4;          // 区号id发
}

message rep_E_LOGIN_MSG_GG2ROOT_LOGIN
{
    int32 isok = 1;
	repeated role_snapshot_t role_list = 2; // 角色快照列表
}

// E_LOGIN_MSG_ROOT2LOGIN_ACCT_CHECK                   = 4,  // root->login账号校验
message req_E_LOGIN_MSG_ROOT2LOGIN_ACCT_CHECK
{
    string account_name = 1;    // 账号名字
    string pswd = 2;            // 密码
    int64 account_id = 3;       // 账号唯一id
}

message rep_E_LOGIN_MSG_ROOT2LOGIN_ACCT_CHECK
{
    int32 isok = 1;
}

// E_LOGIN_MSG_C2S_CREATE_ROLE                         = 5,  // c->s 创建角色
message req_E_LOGIN_MSG_C2S_CREATE_ROLE
{
	string nickname = 1;
}
message rep_E_LOGIN_MSG_C2S_CREATE_ROLE
{
	int32 isok = 1;
	role_snapshot_t snapshot = 2;	// 角色快照
}

// E_LOGIN_MSG_GG2ROOT_CREATE_ROLE                     = 6,  // gg->root 创建角色
message req_E_LOGIN_MSG_GG2ROOT_CREATE_ROLE
{
	int64 acc_id = 1;
	string nickname = 2;
}

message rep_E_LOGIN_MSG_GG2ROOT_CREATE_ROLE
{
	int32 isok = 1;
	role_snapshot_t snapshot = 2;	// 角色快照
}

//    E_LOGIN_MSG_ROOT2LOGIN_CREATE_ROLE                  = 7,  // root->login 创建角色
message req_E_LOGIN_MSG_ROOT2LOGIN_CREATE_ROLE
{
	int64 acc_id = 1;
	string nickname = 2;
}

message rep_E_LOGIN_MSG_ROOT2LOGIN_CREATE_ROLE
{
	int32 isok = 1;
	string nickname = 2;
	int64 acc_id = 3;
	int64 role_id = 4;
}

// E_LOGIN_MSG_ROOT2DATA_CREATE_ROLE                   = 8,  // root->data 创建角色
message req_E_LOGIN_MSG_ROOT2DATA_CREATE_ROLE
{
	int64 acc_id = 1;
	string nickname = 2;
	int64 role_id = 3;
}
message rep_E_LOGIN_MSG_ROOT2DATA_CREATE_ROLE
{
	int32 isok = 1;
}
   
// E_LOGIN_MSG_ROOT2LOGIC_CREATE_ROLE                  = 9,  // root->logic 创建角色
message req_E_LOGIN_MSG_ROOT2LOGIC_CREATE_ROLE
{
	int64 role_id = 1;
	int64 acc_id = 2;
	string nickname = 3;
}

message rep_E_LOGIN_MSG_ROOT2LOGIC_CREATE_ROLE
{
	int32 isok = 1;
	role_snapshot_t snapshot = 2;	// 角色快照
}

// E_LOGIN_MSG_ROOT2DATA_FETCH_ROLE_SNAPSHOT           = 10, // 获取玩家角色快照
message req_E_LOGIN_MSG_ROOT2DATA_FETCH_ROLE_SNAPSHOT
{
	int64 acc_id = 1;	// 账号id
}

message rep_E_LOGIN_MSG_ROOT2DATA_FETCH_ROLE_SNAPSHOT
{
	int32 isok = 1;
	repeated role_snapshot_t role_list = 2; // 角色快照列表
}

// E_LOGIN_MSG_C2S_ENTER_GAME                          = 11,  // 客户端请求进入游戏
message req_E_LOGIN_MSG_C2S_ENTER_GAME
{
	int64 role_id = 1;
}

message rep_E_LOGIN_MSG_C2S_ENTER_GAME
{
	int32 isok = 1;
	role_game_data game_data = 2;
}

// E_LOGIN_MSG_C2S_TEST_ROLE_MSG                       = 12,  // 测试角色相关的消息
message req_E_LOGIN_MSG_C2S_TEST_ROLE_MSG
{
	string echo_content = 1;
}

message rep_E_LOGIN_MSG_C2S_TEST_ROLE_MSG
{
	int32 isok = 1;
	string echo_content = 2;
}
// ------------------------------------------------ server msg --------------------------------------------


// E_LOGIN_MSG_GG2CENTER_ENTER_GAME                    = 101,   // gate->center 进入游戏
message req_E_LOGIN_MSG_GG2CENTER_ENTER_GAME
{
	int64 role_id = 1;
	int64 acc_id = 2;
}

message rep_E_LOGIN_MSG_GG2CENTER_ENTER_GAME
{
	int32 isok = 1;
	int32 logic_index = 2;		// 逻辑进程index
}

// E_LOGIN_MSG_CENTER2ROOT_ENTER_GAME                  = 102,   // center->root 玩家进入游戏
message req_E_LOGIN_MSG_CENTER2ROOT_ENTER_GAME
{
	int64 role_id = 1;
	int64 acc_id = 2;
	int32 gate_index = 3;
}

message rep_E_LOGIN_MSG_CENTER2ROOT_ENTER_GAME
{
	int32 isok = 1;
	int32 logic_index = 2;	// logic_server的节点索引
}

// E_LOGIN_MSG_ROOT2LOGIC_ENTER_GAME                   = 103,   // root->logic 玩家进入游戏
message req_E_LOGIN_MSG_ROOT2LOGIC_ENTER_GAME
{
	int64 role_id = 1;
	int64 acc_id = 2;
	int32 gate_index = 3;
}
message rep_E_LOGIN_MSG_ROOT2LOGIC_ENTER_GAME
{
	int32 isok = 1;
}

// E_LOGIN_MSG_GATE2LOGIC_ENTER_GAME                   = 104,   // gate->logic 玩家进入游戏
message req_E_LOGIN_MSG_GATE2LOGIC_ENTER_GAME
{
	int64 role_id = 1;
	int64 acc_id = 2;
}

message rep_E_LOGIN_MSG_GATE2LOGIC_ENTER_GAME
{
	int32 isok = 1;
	role_game_data game_data = 2;
}